packages/cli/src/commands/integration/remove.ts

```diff
@@ -0,0 +1,357 @@
+): Promise<number> {
+  client.output.spinner('Retrieving resource…', 500);
+  const resources = await getResources(client, team.id);
+  const targetedResource = resources.find(
+    resource => resource.name === resourceName
+  );
+
+  if (!targetedResource) {
+    client.output.error(`No resource ${chalk.bold(resourceName)} found.`);
+    return 0;
+  }
+
+  if (args.length === 3) {
+    if (flags['--unlink-all']) {
+      client.output.error(
+        'Cannot specify a project when using the `--unlink-all` flag.'
+      );
+      return 1;
+    }
+
+    const unlinkProjectResults = await handleUnlinkProject(
+      client,
+      targetedResource,
+      args[2],
+      !!flags['--yes']
+    );
+    if (unlinkProjectResults !== undefined) {
+      return unlinkProjectResults;
+    }
+  } else if (flags['--unlink-all']) {
+    const unlinkAllResults = await handleUnlinkAllProjects(
+      client,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (unlinkAllResults !== undefined) {
+      return unlinkAllResults;
+    }
+  }
+
+  if (flags['--delete']) {
+    const deleteResult = await handleDeleteResource(
+      client,
+      team,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (deleteResult !== undefined) {
+      return deleteResult;
+    }
+  }
+
+  return 0;
+}
+
+// $ vc integration remove [resource-name] [project]
+async function handleUnlinkProject(
+  client: Client,
+  resource: Resource,
+  projectName: string,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  const project = resource.projectsMetadata?.find(
+    project => projectName === project.name
+  );
+  if (!project) {
+    client.output.log(
+      `Could not find project ${chalk.bold(projectName)} linked to resource ${chalk.bold(resource.name)}.`
+    );
+    return 0;
+  }
+
+  if (
+    !skipConfirmation &&
+    !(await confirmUnlinkProject(client, resource, project))
+  ) {
+    client.output.log('Canceled');
+    return 0;
+  }
+
+  try {
+    client.output.spinner('Unlinking resource…', 500);
+    await disconnectResourceFromProject(client, resource, project);
+    client.output.success(
+      `Unlinked ${chalk.bold(project.name)} from ${chalk.bold(resource.name)}`
+    );
+    resource.projectsMetadata = resource.projectsMetadata?.filter(
+      project => projectName !== project.name
+    );
+  } catch (error) {
+    client.output.error(
+      `A problem occurred while unlinking: ${(error as Error).message}`
+    );
+    return 1;
+  }
+
+  return;
+}
+
+// $ vc integration remove [resource-name] --unlink-all
+async function handleUnlinkAllProjects(
+  client: Client,
+  resource: Resource,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  if (resource.projectsMetadata?.length === 0) {
+    client.output.log(
+      `${chalk.bold(resource.name)} has no projects to unlink.`
+    );
+  } else {
+    if (
+      !skipConfirmation &&
+      !(await confirmUnlinkAllProjects(client, resource))
```

Comment:
seems to be used quite often, perhaps wrap in a method and reuse in all your calls?

Reply:
Not quite, each confirm is slightly different, and each one is only called once. This is me wrapping them for better readability.